### PR TITLE
Topic/bounded controls: cleaned up branch ready.

### PR DESCRIPTION
**Pull request by telephon**
This is a cleaned up version of the old branch bounded-controls. It fixes #1252.

### CODE REVIEW

**Comment by telephon** [comment]
ping?

File: server/scsynth/SC_GraphDef.cpp [hunk 1]
```diff
@@ -246,6 +246,60 @@ void GraphDef_ReadVariant(World *inWorld, char*& buffer, GraphDef* inGraphDef, G
 	}
 }
 
+
+typedef struct IndexMap {
+	uint32 index;
+	uint32 mIndex;
```
Comment (timblechmann) [inline server/scsynth/SC_GraphDef.cpp]: what is index, what is mIndex? m is a common prefix for "member", though both are members. maybe there are more expressive names?

**Comment by telephon** [comment]
better?

**Comment by timblechmann** [comment]
great, thanks!

**Comment by telephon** [comment]
if it is ok, could someone else be so kind and merge?

**Comment by telephon** [comment]
Ok, I'll take responsibility, before it gets stale.

**Comment by LFSaw** [comment]
thanks for all the effort to merge in topics, @telephon !

**Pull request merged**